mandatory/3-playing-computer.js

```diff
@@ -7,36 +7,47 @@
   Answer the following questions:
 
   1. This program throws an error. Why? (If you can't find it, try executing it).
-  2. Remove the line that throws the error.
+    [ console.log(b) throws error as b is undefined]
+  2. Remove the line that throws the error. 
+    [commented out the line causing errors]
   3. What is printed to the console?
+    [2, 6, 6, 4, 7, 9, 8, 6, 9, 13, 10, 8]
   4. How many times is "f1" called?
+    [10 times, used console.log("find", a)]
   5. How many times is "f2" called?
+    [2 times, used console.log("search", x)]
   6. What value does the "a" parameter take in the first "f1" call?
+    [value a = 9, used console.log("before", a)]
   7. What is the value of the "a" outer variable when "f1" is called for the first time?
+    [value a = 10, used console.log("after", a)]
```

Comment:
The value of 'a' is 8. The value of 'a' is initially 6. The f1 function first gets called on the second iteration of the for loop in which time the 'a' variable has incremented by 1 twice. to give a value of 8.